Make the port build with FFMPEG 3.x.

Obtained from: http://anonscm.debian.org/viewvc/pkg-kde/kde-extras/strigi/trunk/debian/patches/ffmpeg_2.9.patch?view=markup&pathrev=20374

Description: Replace deprecated FFmpeg API
Author: Andreas Cadhalpun <Andreas.Cadhalpun@googlemail.com>
Last-Update: <2015-11-02>
--- libstreamanalyzer/plugins/endplugins/ffmpegendanalyzer.cpp.orig
+++ libstreamanalyzer/plugins/endplugins/ffmpegendanalyzer.cpp
@@ -355,7 +355,7 @@ FFMPEGEndAnalyzer::analyze(AnalysisResul
 #endif
 
   if(fc->bit_rate)
-    ar.addValue(factory->bitrateProperty, fc->bit_rate);
+    ar.addValue(factory->bitrateProperty, (uint32_t)fc->bit_rate);
   else if (fc->duration!= no_bitrate && fc->duration > 0) {
     cout<<"Trying to estimate bitrate\n";
     int64_t size;
@@ -412,8 +412,8 @@ FFMPEGEndAnalyzer::analyze(AnalysisResul
         if (size_t len = strlen(p->name)) {
           ar.addTriplet(streamuri, codecPropertyName, string(p->name, len));
         }
-      } else if (size_t len = strlen(codec.codec_name)) {
-        ar.addTriplet(streamuri, codecPropertyName, string(codec.codec_name, len));
+      } else if (size_t len = strlen(avcodec_get_name(codec.codec_id))) {
+        ar.addTriplet(streamuri, codecPropertyName, string(avcodec_get_name(codec.codec_id), len));
       }
 /*
 00792     } else if (enc->codec_id == CODEC_ID_MPEG2TS) {
@@ -486,7 +486,7 @@ FFMPEGEndAnalyzer::analyze(AnalysisResul
           outs << stream.avg_frame_rate.num / stream.avg_frame_rate.den;
           ar.addTriplet(streamuri, frameRatePropertyName, outs.str());
         }
-        if (codec.pix_fmt != PIX_FMT_NONE) {}//FIXME pixel format
+        if (codec.pix_fmt != AV_PIX_FMT_NONE) {}//FIXME pixel format
       }
       
     }
